packages/react/src/primitives/shared/responsive/__tests__/useBreakpoint.test.ts

```diff
@@ -69,7 +48,53 @@ describe('useBreakpoint', () => {
     expect(breakpoint).toBe('base');
   });
 
-  // Test going down and up the breakpoints
-  mediaQueries.forEach(testBreakpoints);
-  mediaQueries.reverse().forEach(testBreakpoints);
+  it.each(mediaQueries)(
+    `should return the $breakpoint breakpoint as expected`,
+    ({ query, breakpoint }) => {
+      matchMedia.useMediaQuery(query);
+      const { result } = renderHook(() =>
+        useBreakpoint({ breakpoints, defaultBreakpoint })
+      );
+      const resultBreakpoint = result.current;
+
+      expect(resultBreakpoint).toBe(breakpoint);
+      expect(mockUseDebugValue).toHaveBeenLastCalledWith(
+        breakpoint,
+        expect.any(Function)
+      );
+
+      if (breakpoint === 'base') {
+        // eslint-disable-next-line jest/no-conditional-expect
```

Comment:
Disabled `jest/no-conditional-expect` here (and below on lines 70, 92, and 95) to avoid refactoring further, which most likely require moving away from the current approach of using the iterative `it.each`